src/hiveboard_bridge/include/hiveboard_bridge/StreamListener.h

```diff
@@ -0,0 +1,32 @@
+#ifndef CATKIN_ROS_STREAMLISTENER_H
+#define CATKIN_ROS_STREAMLISTENER_H
+
+#include "ReceiveThreadAction.h"
+#include "ros/ros.h"
+#include <common/IProtobufStream.h>
+#include <functional>
+#include <hivemind-host/MessageDTO.h>
+#include <optional>
+#include <thread>
+
+/**
+ * A class that listens on a stream and performs some actions when messages arrive
+ */
+class StreamListener {
```

Comment:
From what I see it doesn't listen to a stream at all, it just creates a thread from a threadaction and loops it's action

Reply:
Right. There has been a lot of moving stuff around so this class is now more generic. I renamed it for `ThreadWrapper`, since this is what this class essentially does.